src/services/FileService.ts

```diff
@@ -0,0 +1,271 @@
+    file.size = size;
+    file.userId = userId;
+    if (mimeType) {
+      file.mimeType = mimeType;
+    }
+    
+    await services.em.persistAndFlush(file);
+    return file;
+  }
+
+  /**
+   * Get file by ID and user ID
+   */
+  async getFileById(fileId: number, userId: number): Promise<File | null> {
+    const { services } = await this.getServices();
+    
+    return await services.file.findOne({ id: fileId, userId });
+  }
+
+  /**
+   * Update file (name and path only - virtual changes)
+   */
+  async updateFile(
+    fileId: number, 
+    userId: number, 
+    name?: string, 
+    path?: string
+  ): Promise<File | null> {
+    const { services } = await this.getServices();
+    
+    const file = await this.getFileById(fileId, userId);
+    if (!file) {
+      return null;
+    }
+
+    if (name !== undefined) {
+      file.name = name;
+    }
+
+    if (path !== undefined) {
+      this.validatePath(path);
+      file.path = path;
+    }
+
+    await services.em.flush();
+    return file;
+  }
+
+  /**
+   * Delete file (removes from S3 and database)
+   */
+  async deleteFile(fileId: number, userId: number): Promise<boolean> {
+    const { services } = await this.getServices();
+    
+    const file = await this.getFileById(fileId, userId);
+    if (!file) {
+      return false;
+    }
+
+    try {
+      // Delete from S3
+      await this.getMinioService().deleteFile(this.bucketName, file.objectKey);
+      
+      // Delete from database
+      await services.em.removeAndFlush(file);
+      return true;
+    } catch (error) {
+      console.error('Error deleting file:', error);
+      return false;
+    }
+  }
+
+  /**
+   * List files by directory with pagination
+   */
+  async listFiles(
+    userId: number,
+    path?: string,
+    page: number = 1,
+    limit: number = 20
+  ): Promise<Page<File>> {
+    const { services } = await this.getServices();
+    
+    const offset = (page - 1) * limit;
+
+    // Build where clause
+    const whereClause: any = { userId };
+    if (path) {
+      whereClause.path = path;
+    }
+
+    const findAndCount = await services.file.findAndCount(whereClause, {
+      orderBy: { createdAt: 'DESC' },
+      limit,
+      offset,
+    });
+
+    return toPageDTO(findAndCount, page, limit);
+  }
+
+  /**
+   * Get all files in a directory tree (recursive)
+   */
+  async getDirectoryTree(userId: number, path: string = '/'): Promise<File[]> {
+    const { services } = await this.getServices();
+    
+    // Get all files that start with the given path
+    const files = await services.file.find({
+      userId,
+      path: { $like: `${path}%` }
+    }, {
+      orderBy: { path: 'ASC', name: 'ASC' }
+    });
+
+    return files;
+  }
+
+  /**
+   * Create a directory (virtual - just a file with no size)
+   */
+  async createDirectory(name: string, path: string, userId: number): Promise<File> {
+    const { services } = await this.getServices();
+    
+    this.validatePath(path);
```

Comment:
Create an empty for directory placeholder is a good ideal .But you can consider this simplier solution: If user create a folder in frontend and don't upload anything this folder will not created. If you  still want to do this way, you must create a flag in the File.ts entity to mark it as a placeholder file for an empty folder and don't return this file when frontend try to query the list file of this empty folder. Chose your preferer solution both soud good for me